### ISSUE TITLE
File hash with absolute paths

### ISSUE THREAD

**Issue by madwizard-thomas** [jakubpawlowicz/clean-css#868]
First of all thanks for your work on clean-css 4.0! Glad it's stable now.

When trying out the new clean-css for a gulp plugin I ran into an issue when using absolute paths (on windows) as keys for the file hash passed to minify.

Say you have the file c:\test\a.css (contents p {color:blue} ) and run the following with c:\test\ as the current working directory:
```javascript
cleanCSS.minify({'a.css' : { styles : "p {color:red}" }});(url)
```
This works fine, however using an absolute path (which seemed possible in earlier clean-css versions):
```javascript
cleanCSS.minify({'c:\\test\\a.css' : { styles : "p {color:red}" }});(url)
```
It will run successfully but actually use the contents of the physical c:\test\a.css file rather than the provided contents (i.e. output blue instead of red).

Also, when using a path 'b.css', which doesn't exist works fine as 'b.css', but when 'c:\\test\\b.css' is used it fails with "Ignoring local \@import of "b.css" as resource is missing.'.'"

Note that it will work if d:\\test\\b.css (different drive) is used.

The problem appears to be caused by the path being stored in inlinerContext.externalContext.sourcesContent. In the fromHash function, the user specified path is passed through normalizePath (producing c:/test/a.css). However, in inlineLocalStylesheet first relativeToCurrentPath is calculated ('a.css'), this path is then normalized (stays the same 'a.css'). This final path 'a.css' will not match the key c:/test/a.css' in sourcesContent. It will fallback to the physical file system and for a.css load the file from disk (which does not have to be the same in gulp context), or fail if no file exists with the given name.

I don't know if passing absolute file paths as keys is officially supported but this is the easiest way to pass a file path when using gulp vinyl files and ensure everything will be rebased correctly.

**Comment by jakubpawlowicz** [comment]
Using absolute paths should work fine (there are tests for it) - thanks for digging into reader sources 👍  I'll be able to investigate it further tomorrow.

**Comment by jakubpawlowicz** [comment]
^ should be it - I'm gonna test it a bit more so it should ship with 4.0.1 tomorrow.

**Comment by jakubpawlowicz** [comment]
Fixed in 4.0.1.
